src/components/helpers/highlight_hook.ts

```diff
@@ -0,0 +1,29 @@
+import { onMounted, useEffect, useEnv } from "@odoo/owl";
+import { useLocalStore } from "../../store_engine";
+import { HighlightProvider, HighlightStore } from "../../stores/highlight_store";
+import { Ref, SpreadsheetChildEnv } from "../../types";
+import { useHoveredElement } from "./listener_hook";
+
+export function useHighlightsOnHover(ref: Ref<HTMLElement>, highlightProvider: HighlightProvider) {
+  const hoverState = useHoveredElement(ref);
+  const env = useEnv() as SpreadsheetChildEnv;
+
+  useHighlights({
+    get highlights() {
+      return hoverState.hovered ? highlightProvider.highlights : [];
+    },
+  });
+  useEffect(
+    () => {
+      env.model.dispatch("RENDER_CANVAS");
+    },
```

Comment:
in this case, I wonder why we still need a command that will be handled by only one component and have no side effects (history). Should we not have a simple function call to model.drawGrid() or useGridDrawing ?

Reply:
you're right, we don't need this command. It's only dispatched to trigger a rendering. Which can be achieved with a simple function call :+1:

Reply:
mouais, not that straight forward because it doesn't have a direct access to the canvas